packages/data-point-codemods/transforms/reducer-args-acc-to-val-acc.js

```diff
@@ -0,0 +1,160 @@
+const util = require('util')
+const _ = require('lodash')
+
+module.exports = (file, api, options) => {
+  const j = api.jscodeshift
+  const root = j(file.source)
+
+  function replaceWith (target, newObject) {
+    for (let key in target) {
+      delete target[key]
+    }
+    Object.assign(target, newObject)
+  }
+
+  function addValueParam (node) {
+    const accReferences = j(node.value.body)
+      .find(j.Identifier, {
+        name: 'acc'
+      })
+      .size()
+
+    let injectAcc = node.value.params.length === 2 || accReferences > 0
+
+    const newParams = [
+      {
+        type: 'Identifier',
+        name: 'input'
+      }
+    ]
+
+    node.value.params = injectAcc
+      ? newParams.concat(node.value.params)
+      : newParams
+  }
+
+  function replaceAccValueReferences (node) {
```

Comment:
Do I understand this right: If the function uses `acc` to reference the accumulator, it'll change, but if they've used `a` or `accumulator` then this won't work? Is it possible to only check the arity and update accordingly?

Also, what happens with something like this?

```js
// increments
dataPoint.transform(a => a.value++)
```

Reply:
- it complicates the code, not that easy to tell which are reducers and which arent, 
- .transform I have a check to ignore those handlers

Reply:
OK, fair points. Thanks for clarifying.